tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/Session.java

```diff
@@ -141,6 +141,46 @@ public void close() {
     }
   }
 
+  /**
+   * Execute the graph's initializers.
+   *
+   * <p>This method is equivalent to {@code session.run(Ops.create(session.graph).init())}.
+   */
+  public void initialize() {
+    synchronized (initLock) {
+      if (hasInitialized) {
+        throw new IllegalStateException("Session has already been initialized.");
+      }
+
+      if (!graph.hasInitializers()) {
+        hasInitialized = true;
+        return;
+      }
+
+      List<Operation> initializers = graph.initializers();
+      if (!initializers.isEmpty()) {
+        Runner runner = runner();
+        initializers.forEach(runner::addTarget);
+        runner.runNoInit();
+      }
+      hasInitialized = true;
+    }
+  }
+
+  /** Create a session and initialize it. */
+  public static Session initialized(Graph graph) {
```

Comment:
These methods should have a better name. Maybe `createAndInitializeSession`? Though that's rather verbose.

Reply:
What do you think about making it part of the constructor by default, but having a `Session(Graph graph, boolean initialize)` so that it can be disabled?  We're doing native calls anyways, and it should avoid people running into the init check by accident.

Reply:
Another option would be to by default, run `init` before the first `run` call, with a boolean to disable doing that.

Reply:
I think having a boolean `initialize` in the constructor sounds good, but let's keep the version without it with a default value to false, to mimic actual behavior?